framework/core/src/Foundation/ApplicationInfoProvider.php

```diff
@@ -0,0 +1,205 @@
+ */
+
+namespace Flarum\Foundation;
+
+use Carbon\Carbon;
+use Flarum\Locale\Translator;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Flarum\User\SessionManager;
+use Illuminate\Console\Scheduling\Schedule;
+use Illuminate\Contracts\Queue\Queue;
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use InvalidArgumentException;
+use SessionHandlerInterface;
+
+class ApplicationInfoProvider
+{
+    /**
+     * @var SettingsRepositoryInterface
+     */
+    protected $settings;
+
+    /**
+     * @var Translator
+     */
+    protected $translator;
+
+    /**
+     * @var Schedule
+     */
+    protected $schedule;
+
+    /**
+     * @var ConnectionInterface
+     */
+    protected $db;
+
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @var SessionManager
+     */
+    protected $session;
+
+    /**
+     * @var SessionHandlerInterface
+     */
+    protected $sessionHandler;
+
+    /**
+     * @param SettingsRepositoryInterface $settings
+     * @param Translator $translator
+     * @param Schedule $schedule
+     * @param ConnectionInterface $db
+     * @param Config $config
+     * @param SessionManager $session
+     * @param SessionHandlerInterface $sessionHandler
+     */
+    public function __construct(
+        SettingsRepositoryInterface $settings,
+        Translator $translator,
+        Schedule $schedule,
+        ConnectionInterface $db,
+        Config $config,
+        SessionManager $session,
+        SessionHandlerInterface $sessionHandler
+    ) {
+        $this->settings = $settings;
+        $this->translator = $translator;
+        $this->schedule = $schedule;
+        $this->db = $db;
+        $this->config = $config;
+        $this->session = $session;
+        $this->sessionHandler = $sessionHandler;
+    }
+
+    /**
+     * Identify if any tasks are registered with the scheduler.
+     *
+     * @return bool
+     */
+    public function scheduledTasksRegistered(): bool
+    {
+        return count($this->schedule->events()) > 0;
+    }
+
+    /**
+     * Gets the current status of the scheduler.
+     *
+     * @return string
+     */
+    public function getSchedulerStatus(): string
+    {
+        $status = $this->settings->get('schedule.last_run');
+
+        if (! $status) {
+            return $this->translator->trans('core.admin.dashboard.status.scheduler.never-run');
+        }
+
+        // If the schedule has not run in the last 5 minutes, mark it as inactive.
+        return Carbon::parse($status) > Carbon::now()->subMinutes(5)
+            ? $this->translator->trans('core.admin.dashboard.status.scheduler.active')
+            : $this->translator->trans('core.admin.status.scheduler.inactive');
+    }
+
+    /**
+     * Identify the queue driver in use.
+     *
+     * @param Queue $queue
+     * @return string
+     */
+    public function identifyQueueDriver(Queue $queue): string
```

Comment:
Can we inject queue instead of requiring it as a parameter?

Reply:
You know, I saw that and thought that could be changed. No idea why I didn't actually do it! Anyway, now refactored :)